tools/ci/cmake-linter.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+ICEORYX_ROOT_PATH=$(git rev-parse --show-toplevel)
+EXIT_CODE=0
+
+performCmakeLinting()
```

Comment:
I completely agree. I checked this out: https://github.com/cmake-lint/cmake-lint and it would be interesting to integrate this into our pipeline.
The goal of this script is to prohibit the use of `add_library` and `add_executable` and force the use of `iox_add_library` and `iox_add_executable` something which I could not realize with cmake-lint. So I would stick to this script.

The reason to prohibit this is, that when using the `add_` commands manually I encountered several issues throughout the code where the C++ standard was missing, position independent code was not set and in the future some may forget to set the rpath correctly or we may have inconsistency with installation directives.

Reply:
@elfenpiff @mossmaurice It is good to have this check for now so that one does not accidently use the wrong functions in CMake. Nevertheless to avoid scope creep in this PR the cmake-lint topic is addressed here as a follow-up.